src/ipc-frontend-dbus.c

```diff
@@ -0,0 +1,698 @@
+ * to the client and FALSE is returned to the caller.
+ *
+ * Returns FALSE on error, TRUE otherwise.
+ */
+static gboolean
+generate_id_pid_mix_from_invocation (IpcFrontendDbus        *self,
+                                     GDBusMethodInvocation *invocation,
+                                     guint64               *id,
+                                     guint64               *id_pid_mix)
+{
+    gboolean pid_ret = FALSE;
+    guint32  pid = 0;
+
+    pid_ret = get_pid_from_dbus_invocation (self->dbus_daemon_proxy,
+                                            invocation,
+                                            &pid);
+    if (pid_ret == TRUE) {
+        *id = random_get_uint64 (self->random);
+        *id_pid_mix = *id ^ pid;
+    } else {
+        g_dbus_method_invocation_return_error (invocation,
+                                               TABRMD_ERROR,
+                                               TABRMD_ERROR_INTERNAL,
+                                               "Failed to get client PID");
+    }
+
+    return pid_ret;
+}
+/*
+ * Mix PID into provide id. Returns mixed value in id_pid_mix out parameter.
+ * NOTE: if an error occurs then an error response is sent through the
+ * invocation to the client and FALSE is returned to the caller
+ *
+ * Returns FALSE on error, TRUE otherwise.
+ */
+static gboolean
+get_id_pid_mix_from_invocation (GDBusProxy            *proxy,
+                                GDBusMethodInvocation *invocation,
+                                guint64                id,
+                                guint64               *id_pid_mix)
+{
+    guint32 pid = 0;
+    gboolean pid_ret = FALSE;
+
+    g_debug ("get_id_pid_mix_from_invocation");
+    pid_ret = get_pid_from_dbus_invocation (proxy,
+                                            invocation,
+                                            &pid);
+    g_debug ("id 0x%" PRIx64 " pid: 0x%" PRIx32, id, pid);
+    if (pid_ret == TRUE) {
+        *id_pid_mix = id ^ pid;
+        g_debug ("mixed: 0x%" PRIx64, *id_pid_mix);
+    } else {
+        g_dbus_method_invocation_return_error (
+            invocation,
+            TABRMD_ERROR,
+            TABRMD_ERROR_INTERNAL,
+            "Failed to get client PID");
+    }
+
+    return pid_ret;
+}
+/*
+ * This is a signal handler for the handle-create-connection signal from
+ * the Tpm2AccessBroker DBus interface. This signal is triggered by a
+ * request from a client to create a new connection with the tabrmd. This
+ * requires a few things be done:
+ * - Create a new ID (uint64) for the connection.
+ * - Create a new Connection object getting the FDs that must be returned
+ *   to the client.
+ * - Build up a dbus response to the client with their connection ID and
+ *   send / receive FDs.
+ * - Send the response message back to the client.
+ * - Insert the new Connection object into the ConnectionManager.
+ * - Notify the CommandSource of the new Connection that it needs to
```

Comment:
This action has been removed from the code, so the description should be deleted too.

Reply:
This whole comment needed to be cleaned up. Thanks for pointing this out.